src/native/crt.c

```diff
@@ -265,6 +265,15 @@ bool aws_jni_check_and_clear_exception(JNIEnv *env) {
     return exception_pending;
 }
 
+jthrowable aws_jni_get_and_clear_exception(JNIEnv *env) {
+    jthrowable exception = (*env)->ExceptionOccurred(env);
+    if (exception) {
+        (*env)->ExceptionDescribe(env);
+        (*env)->ExceptionClear(env);
+    }
+    return exception;
```

Comment:
this is probably going to lead to user-after-free

I assume ExceptionClear() will lead to the exception being garbage-collected, but we're returning its pointer.

This code worked in `aws_jni_check_and_clear_exception()` because we were just returning a `bool` of whether or not the pointer was non-null